tests/flow/test_graph_merge.py

```diff
@@ -592,3 +592,29 @@ def test28_merge_reset_label_scan(self):
         query = """MERGE (a:Person {name: 'a'})-[:FRIEND]->() WITH NULL AS a MATCH (n1:Person), (n2:Person) MERGE (:NEW)"""
         result = graph.query(query)
         self.env.assertEquals(result.nodes_created, 1)
+
+    def test29_merge_resue(self):
+        redis_con = self.env.getConnection()
+        graph = Graph(redis_con, "merge_reuse")
+        query = """
+        CREATE (m:L1 {v: "abc"})
+        CREATE (u:L2 {v: "x"})
+        CREATE (n:L2 {v: "y"})
+        CREATE (:L2 {v: 'y'})
+        CREATE (u)-[:R]->(m), (u)-[:R]->(m)
+        CREATE (n)-[:R]->(m), (n)-[:R]->(m)"""
+        graph.query(query)
+
+        query = """
+        MERGE (m:L1 {v: "abc"})
+        SET m.v = "abcd"
+        WITH m
+        MATCH (u:L2 {v: "x"})
+        MATCH (n:L2 {v: "y"})
+        MERGE (u)-[:R]->(m)<-[:R]-(n)
```

Comment:
This query doesn't create any entities, as a result (I believe) it doesn't call propagate reset.
I was under the impression we wanted to exercise the propagate reset functions

Reply:
it is calling propagate reset validated the crash on master before